flake.nix

```diff
@@ -0,0 +1,140 @@
+{
+    description = "Salz";
+    inputs = {
+        nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    };
+
+    outputs = { self, nixpkgs, flake-utils }:
+        let
+            forAllSystems = function:
+                nixpkgs.lib.genAttrs [
+                    "x86_64-linux"
+                    "aarch64-linux"
+                ] (
+                    system:
+                    function system (import nixpkgs {
+                        inherit system;
+                        config.allowUnfree = true;
+                        overlays = [];
+                    }));
+            forAllSystemsPkgsOnly = function: (forAllSystems (_: pkgs: function pkgs));
+        in {
+            packages = forAllSystemsPkgsOnly (pkgs: {
+                api = pkgs.buildGoModule (finalAttrs: {
+                    pname = "api";
+                    version = "0.0.1";
+                    src = ./api;
+                    vendorHash = "sha256-tgWC6WTyP+juEF1t8yv1ZT4j2O5+YZzwI8JQ9CUHads=";
+                });
+                web = pkgs.buildNpmPackage (finalAttrs: {
+                    pname = "web";
+                    version = "0.0.1";
+                    src = ./web;
+                    npmDepsHash = "sha256-+uK8EW08Cq2hfaQdT/e8cg4SXD7Qixlq9bIs/C5+rWI=";
+                });
+            });
+            devShells = forAllSystems (system: pkgs: {
+                default = pkgs.mkShell {
+                    # Pull in package dependencies
+                    inputsFrom = [
+                        self.packages.${system}.api
+                        self.packages.${system}.web
+                    ];
+
+                    # Development-only packages
+                    packages = [
+                        # Infrastructure
+                        pkgs.git
+                        (pkgs.callPackage ./nix/hcp.nix {})
+                        pkgs.process-compose
+
+                        # Minio
+                        pkgs.minio
+                        pkgs.minio-client
+
+                        # DB
+                        pkgs.postgresql_16
+                        pkgs.sqlfluff
+                        pkgs.jsonschema
+
+                        # Node
+                        pkgs.nodePackages.typescript-language-server
+                        pkgs.nodePackages.svelte-language-server
+                    ];
+                    shellHook = ''
```

Comment:
I don't think I'm a fan of defining all this in the shell hook. It makes more sense for me to have this in process-compose, but maybe because that's because I'm used to the docker compose workflow (environment variables and all the other configuration is defined in the compose file).

Reply:
It does let you reuse a lot of things though (e.g. STATEDIR, the minio access keys, postgres access, etc)

Reply:
Yeah, I put it in the shell hook because:
- dev tools like `mc`, `psql`, etc use the environment variables to talk with services
- we can dynamically change and inspect them as we develop

Reply:
If we do end up using process-compose for initial deployment, then I think it would make sense to setup the env vars in the deployment process-compose config. For development, it seems more convenient to me to have them in the user's env.